### PR TITLE
[narwhal] fix Narwhal Docker image build

### PR DESCRIPTION
**Pull request by akichidis**
When try to build the Narwhal Docker image the following error is produced:
```
 > [planner 3/3] RUN sed -i '/crates\/workspace-hack/b; /crates/d; /narwhal/d' Cargo.toml     && cargo metadata -q >/dev/null:
#15 339.3 error: no matching package found
#15 339.3 searched package name: `move-abigen`
#15 339.3 perhaps you meant:      move-docgen
```

https://github.com/MystenLabs/sui-operations/actions/runs/3297822824/jobs/5439131120

After looking on differences compared to the SUI dockerfile I've identified that we use a different version of the base image and rust tool kit. I've now aligned to what SUI does. In a follow up PR I'll get those Dockerfiles even closer to their functionality. Also a `build.sh` script file (similar to what we use for SUI) has been introduced to help easily build an image by just running `./narwhal/Docker/build.sh`

### CODE REVIEW

File: narwhal/Docker/build.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+# Copyright (c) Mysten Labs, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# fast fail.
+set -e
+
+DIR="$( cd "$( dirname "$0" )" && pwd )"
```
Comment (huitseeker) [inline narwhal/Docker/build.sh:8]: Nit: `DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"` is [more robust](https://stackoverflow.com/questions/35006457/choosing-between-0-and-bash-source).

**Review by huitseeker — APPROVED** [review]
Thanks!

File: narwhal/Docker/build.sh (hunk 1 above)
Comment (akichidis) [reply narwhal/Docker/build.sh:8]: oh nice, thanks for the tip!

File: narwhal/Docker/Dockerfile [hunk 2]
```diff
@@ -6,25 +6,13 @@ ARG BUILD_MODE="release"
...
-    curl \
-    build-essential \
-    libssl-dev \
-    pkg-config \
-    clang \
-    cmake > /dev/null 2>&1 \
-    && rm -rf /var/lib/apt/lists/*
-# Install the fmt
-RUN rustup component add rustfmt
+RUN apt-get update && apt-get install -y cmake clang
```
Comment (mwtian) [inline narwhal/Docker/Dockerfile:14]: Not sure why building / running Narwhal needs cmake and clang, but probably not hurt.

**Review by mwtian — APPROVED** [review]

File: narwhal/Docker/Dockerfile (hunk 2 above)
Comment (bmwill) [reply narwhal/Docker/Dockerfile:14]: rocksdb most likely

Comment (akichidis) [reply narwhal/Docker/Dockerfile:14]: What @bmwill said, if I remember correctly we saw in the past failing to compile successfully with rocksdb without clang

**Pull request merged**